File: examples/python/pii_detection.py
"""
InstaTools API - PII Detection Example

This example demonstrates how to detect personally identifiable information (PII) in text.
"""

import os
import requests

API_KEY = os.getenv('INSTATOOLS_API_KEY', 'YOUR_API_KEY')
BASE_URL = 'https://api.instatools.co'


def detect_pii(text: str, policy_id: str = 'default') -> dict:
    """Detect PII in text content."""
    response = requests.post(
        f'{BASE_URL}/api/v1/detect',
        headers={
            'Authorization': f'Bearer {API_KEY}',
            'Content-Type': 'application/json'
        },
        json={
            'text': text,
            'policyId': policy_id
        }
    )
    response.raise_for_status()
    return response.json()


def detect_pii_bulk(documents: list, policy_id: str = 'default') -> dict:
    """Detect PII in multiple documents (Enterprise tier only)."""
    response = requests.post(
        f'{BASE_URL}/api/v1/detect/bulk',
        headers={
            'Authorization': f'Bearer {API_KEY}',
            'Content-Type': 'application/json'
        },
        json={
            'documents': documents,
            'policyId': policy_id
        }
    )
    response.raise_for_status()
    return response.json()


def main():
    try:
        # Single text PII detection
        sample_text = """
        Customer Information:
        Name: John Doe
        Email: john.doe@example.com
        SSN: 123-45-6789
        Phone: (555) 123-4567
        Credit Card: 4111-1111-1111-1111
        """

        print('Detecting PII in text...')
        result = detect_pii(sample_text)
        print(f'Result: {result}')

        # Display findings
        findings = result.get('findings', [])
        if findings:
            print('\nPII Found:')
            for i, finding in enumerate(findings, 1):
                print(f"  {i}. {finding['type']}: \"{finding['value']}\" (confidence: {finding['confidence']})")

        # Bulk detection example (Enterprise tier)
        print('\nBulk PII detection...')
        bulk_results = detect_pii_bulk([
            {'id': 'doc1', 'text': 'Contact: jane@company.com'},
            {'id': 'doc2', 'text': 'SSN: 987-65-4321'}
        ])
        print(f'Bulk Results: {bulk_results}')

    except requests.HTTPError as e:
        print(f'HTTP Error: {e.response.status_code} - {e.response.text}')
    except Exception as e:
        print(f'Error: {e}')


if __name__ == '__main__':
    main()

File: examples/javascript/pii-detection.js
/**
 * InstaTools API - PII Detection Example
 * 
 * This example demonstrates how to detect personally identifiable information (PII) in text.
 */

const API_KEY = process.env.INSTATOOLS_API_KEY || 'YOUR_API_KEY';
const BASE_URL = 'https://api.instatools.com';

/**
 * Detect PII in text content
 */
async function detectPII(text, policyId = 'default') {
  const response = await fetch(`${BASE_URL}/api/v1/detect`, {
    method: 'POST',
    headers: {
      'Authorization': `Bearer ${API_KEY}`,
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({
      text,
      policyId
    })
  });

  if (!response.ok) {
    const error = await response.json();
    throw new Error(`API Error: ${error.title} - ${error.detail}`);
  }

  return response.json();
}

/**
 * Detect PII in multiple documents (Enterprise tier only)
 */
async function detectPIIBulk(documents, policyId = 'default') {
  const response = await fetch(`${BASE_URL}/api/v1/detect/bulk`, {
    method: 'POST',
    headers: {
      'Authorization': `Bearer ${API_KEY}`,
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({
      documents,
      policyId
    })
  });

  if (!response.ok) {
    const error = await response.json();
    throw new Error(`API Error: ${error.title} - ${error.detail}`);
  }

  return response.json();
}

// Example usage
async function main() {
  try {
    // Single text PII detection
    const sampleText = `
      Customer Information:
      Name: John Doe
      Email: john.doe@example.com
      SSN: 123-45-6789
      Phone: (555) 123-4567
      Credit Card: 4111-1111-1111-1111
    `;

    console.log('Detecting PII in text...');
    const result = await detectPII(sampleText);
    console.log('Detected PII:', JSON.stringify(result, null, 2));

    // Display findings
    if (result.findings && result.findings.length > 0) {
      console.log('\nPII Found:');
      result.findings.forEach((finding, index) => {
        console.log(`  ${index + 1}. ${finding.type}: "${finding.value}" (confidence: ${finding.confidence})`);
      });
    }

    // Bulk detection example (Enterprise tier)
    console.log('\nBulk PII detection...');
    const bulkResults = await detectPIIBulk([
      { id: 'doc1', text: 'Contact: jane@company.com' },
      { id: 'doc2', text: 'SSN: 987-65-4321' }
    ]);
    console.log('Bulk Results:', JSON.stringify(bulkResults, null, 2));

  } catch (error) {
    console.error('Error:', error.message);
  }
}

main();
